### DISCUSSION TITLE
Any way to ruh a script every time something is copied to the clipboard? Perhaps a clipboard system event?

### DISCUSSION THREAD

**Discussion by bigloudjeff** [johnlindquist/kit#933] [Q&A] [answered]
I would like to analyze the clipboard every time something is copied to see if it is text and if it contains any specific URL patterns that I can then do very operations on. I can do this with CopyQ but I would love to be able to stop using CopyQ if ScriptKit can enable something like this. Thanks!

**Comment by johnlindquist** [comment, marked as answer]
So I found this lib: https://www.npmjs.com/package/clipboard-event

But you'll need to re-compile the `.swift` file before you can run it:

https://github.com/sudhakar3697/node-clipboard-event/issues/17

Basically, install swift, then:

```bash
cd ~/.kenv/node_modules/clipboard-event/platform
swiftc -o clipboard-event-handler-mac clipboard-event-handler-mac.swift
```

Then here's your script:

> Note: If you use the install link, you'll need to edit the script to set `// Background: true` again. Kit.app strips metadata like that when installing scripts.

```js
// Name: Testing Monitor Clipboard
// Background: true

import "@johnlindquist/kit"

let clipboardListener = await npm("clipboard-event")

clipboardListener.startListening()

clipboardListener.on("change", async () => {
  let text = await paste()
  await say(text) // THIS WILL GET ANNOYING QUICKLY 🤣 CHANGE ASAP
})
```

[Open testing-monitor-clipboard in Script Kit](https://scriptkit.com/api/new?name=testing-monitor-clipboard&url=https://gist.githubusercontent.com/johnlindquist/31b7ccb32a75a66e85bd7b6249d014ce/raw/1bb762fa45018abc7e05c60a16ed5355b436eead/testing-monitor-clipboard.ts")

**Reply by bigloudjeff** [reply]
Thanks! I will give it a try. It might be good to have a clipboard event as one of the Kit system events so we don't need to go through all of those steps. Just a suggestion. I am being selfish. ;-)

**Reply by johnlindquist** [reply]
Yeah, I'll have to look into it. I think this lib might be better than the approach I use for the clipboard history. Thanks for the nudge.
